test-integration/helpers/README.md

```diff
@@ -0,0 +1,246 @@
+        defaultWaitTime: 3000
+
+      # instantiate with options by calling the base `TestHelper`'s constructor
+      super(testContext, testElementLocator, COMMON_ELEMENTS, calendarOptions)
+
+      # Not commonly needed.
+      # For this particular UI, a planPopup and methods controlling the popup
+      # are helpful to separate from the main parent calendar UI helper methods.
+      # PlanPopupHelper is an extended TestHelper like CalendarHelper is.
+      @setCommonHelper('planPopup', new PlanPopupHelper(@test))
+  ```
+  You can read more about what happens when you instantiate [here](#helper-details).
+
+4. Extend the helper with additional UI process/action methods as needed, using el's element accessor methods:
+
+  ```coffee
+  class CalendarHelper extends TestHelper
+    constructor: ...
+    createNew: (type) =>
+      @waitUntilLoaded()
+
+      @el.addToggle.get().click()
+      ...
+
+    goPerformanceForecast: =>
+      @test.utils.windowPosition.scrollTop()
+      @el.forecastLink.get().click()
+
+    goOpen: (title) =>
+      # wait until the calendar is open
+      @waitUntilLoaded()
+      # TODO: Make this a `data-title` attribute
+      # HACK: Might need to scroll the item to click on into view
+      el = @el.planByTitle.get(title)
+      @test.utils.windowPosition.scrollTo(el)
+      el.click()
+      @test.utils.windowPosition.scrollTop()
+  ```
+
+# Writing a spec
+
+With the UI helper, a spec can look like this:
+
+```coffee
+{describe, User, CourseSelect, Calendar, ReadingBuilder} = require './helpers'
+{expect} = require 'chai'
+
+TEACHER_USERNAME = 'teacher01'
+
+{CalendarHelper} = Calendar
+
+describe 'Calendar and Stats', ->
+
+  beforeEach ->
+    @user = new User(@)
+    @calendar = new CalendarHelper(@)
+    @courseSelect = new CourseSelect(@)
+
+    @user.login(TEACHER_USERNAME)
+
+  @it 'Shows stats for all published plans (readonly)', ->
+    COURSE_CATEGORY = 'BIOLOGY'
+    @courseSelect.goTo(COURSE_CATEGORY)
+    @calendar.waitUntilLoaded()
+
+    @calendar.el.publishedPlan.forEach (plan, index, total) =>
+      console.log 'Opening', COURSE_CATEGORY, index, '/', total
+      plan.click()
+      @calendar.el.planPopup.waitUntilLoaded()
+
+      @calendar.el.planPopup.isPresent().then (isPresent) ->
+        expect(isPresent).to.be.true
+
+      # -- advance/detailed usage --
+      popupTitle = @calendar.el.planPopup.getTitle()
+      planTitle = @calendar.getPlanTitle(plan)
+
+      selenium.promise.all([popupTitle, planTitle]).then ([popupTitle, planTitle]) ->
+        expect(popupTitle).to.equal(planTitle)
+      # /-- advance/detailed usage --
+
+      @calendar.el.planPopup.el.periodReviewTab.forEach (period) ->
+        period.click()
+
+      @calendar.el.planPopup.close()
+
+    @user.goHome()
+```
+
+# Helper Details
+
+From doing the above, if we instantiate CalendarHelper, we get back something that looks like this:
+
+```coffee
+calendarHelper =
+  test: testContext
```

Comment:
What about keeping the `README` to be examples of how to use the helpers and then editing them could be in a `./advanced.md` file?

For example, rather than describe the object this way (which is not easily copy/pasteable & exposes some implementation details), what about something like:

```js
calendar = new helpers.Calendar(this)

calendar.goForecast()
calendar.goStudentScores()
calendar.goPlanByTitle(title)
// This function will block all selenium based promises until
// the loading element is absent before continuing.
calendar.waitUntilLoaded(optionalWaitTime)

// These are elements you can use to test things about the calendar page.
// They are defined by the COMMON_ELEMENTS section in `helpers/calendar.coffee`
calendar.el.forecastLink      // <TestItemHelper>
calendar.el.studentScoresLink // <TestItemHelper>
calendar.el.addToggle         // <TestItemHelper>
```

I think the rest of the items might be too-overwhelming to get an idea of how to use the helpers (maybe a "For creating/editing helpers see `[./advanced.md](./advanced.md)`" link?)

```js
// "The rest of the items":

calendar.test // testContext
calendar.options // options set when the calendar was instantiated

// uses the loadingLocator option
calendar.el.loadingState      // <TestItemHelper>
// from explicitly calling setCommonHelper
calendar.el.planPopup         // <PlanPopupHelper>

// The following are used to attach the helpers that are now on CalendarHelper.el
// They are exposed here for customizing your helper as needed,
// as in this case with the planPopup.
// Usually, you can just rely on the commonElements object.
calendar.setCommonHelper(name, helper)
calendar.setCommonElement(locator, name)
```